api/v1/api.go

```diff
@@ -447,6 +451,9 @@ func (api *API) insertAlerts(w http.ResponseWriter, r *http.Request, alerts ...*
 		if err := a.Validate(); err != nil {
 			validationErrs.Add(err)
 			api.m.Invalid().Inc()
+			if api.alertLCObserver != nil {
```

Comment:
probably we can instance a no op observer so that you don't have to check if this is nil everytime

Reply:
I did it this way because there are cases where calling the observer method is not free. Like when an alert is added to the aggregation group we compute the `GroupKey` of the aggregation group and pass it in the method. Checking for `nil` helps us avoid this unnecessary calls when its not really providing any value. That is my only reasoning behind this, but if you think otherwise I can change it and use noop as default.